### DISCUSSION TITLE
Can't get `invalidate()` to work but `invalidateAll()` (with StackBlitz demo)

### DISCUSSION THREAD

**Discussion by 205g0** [sveltejs/kit#8975] [Q&A] [answered]
Go to the Stackblitz, go to the test page if not there yet and click both buttons: The `invalidateAll()` works while the other doesn't: https://stackblitz.com/edit/sveltejs-kit-template-default-dagywb?file=src/routes/test/Button.svelte

**Comment by 205g0** [comment, marked as answer]
I think I found the culprit: The url needs to be called within the load function with this special fetch (A load function depends on url if it calls fetch(url) or depends(url)). So, I need to use depends() then if I don't rely on fetch.

Now, I didn't get these custom identifiers for depends(), what's the best structure/syntax to use them?

**Reply by geoffrich** [reply]
Have you seen the docs on [manual invalidation with depends](https://kit.svelte.dev/docs/load#invalidation-manual-invalidation)? Is there something in the example that could be clearer?

**Reply by 205g0** [reply]
Thanks, and I've read it few times haha. To your question:

1. Following sentence should be in bold/red/h1 because it's easily overlooked: ___A load function depends on url if it calls fetch(url) or depends(url).___ I've read the entire section many times without getting that it's not the URL of my app but the one from the inside fetch (I don't have any because I access the db directly in `+page.server.ts`. So, it's not really intuitive. Maybe the `invalidate()` parameter should be called `fetchURLInLoad` or so to make clear it's not the router's url...
2. Then, `depends()`: I am now thinking what should be the best custom identifier structure, I'd love to use just my router structure or should I better not and create use some custom syntax on-top? Both are not optimal, first isn't DRY and second creates decision fatigue and another mental concept on-top on the router structure. Then, (minor), I would call `depends()` rather `endpointId()`, or `loadId` I don't understand the lingual semantic behind depends. `invalidate()` invalidates date from an endpoint with the id xy. `depends` as a word doesn't make any sense to me in that context, does it mean "depends on that id"? it shouldn't because that load function has that id with which its data can be invalidated; edit: or maybe it means it depends on some data, ok then it makes sense again, load function depends on the data car with the id `app:car`, hm...
3. Then, why can't we just put the url of the path of the load function we'd like to invalidate? We want to invalidate data from some `+page.ts`/`+page.server.ts` and it shouldn't matter if they use an inside fetch or not.

Maybe I got all wrong but the whole `invalidate()` api creates more mental drain than I've expected. 🙂
